src/dbnode/storage/bootstrap/bootstrapper/peers/source.go

```diff
@@ -567,43 +596,25 @@ func (s *peersSource) flush(
 				continue
 			}
 
-			flushCtx.Reset()
-			stream, err := bl.Stream(flushCtx)
+			checksum, err := bl.Checksum()
```

Comment:
I don't follow the impact of these 3 changes that seem to reorder operations a bit

Reply:
+1

Reply:
+2

Reply:
So now we aren't doing `.Stream()` anymore and we are discarding the block (which is good because `.Stream()` is expensive), we need to do `.Checksum()` before (the block will already have the pre-calculated checksum).